lxd/containers_post.go

```diff
@@ -432,6 +432,26 @@ func containersPost(d *Daemon, r *http.Request) Response {
 		return BadRequest(fmt.Errorf("Invalid container name: '%s' is reserved for snapshots", shared.SnapshotDelimiter))
 	}
 
+	// Check that there are no contradicting root disk devices.
+	var profileRootDiskDevices []string
+	for _, pName := range req.Profiles {
+		_, p, err := dbProfileGet(d.db, pName)
+		if err != nil {
+			continue
+		}
+
+		v := containerGetRootDiskDevice(p.Devices)
+		if v != nil && !shared.StringInSlice(v["pool"], profileRootDiskDevices) {
+			profileRootDiskDevices = append(profileRootDiskDevices, v["pool"])
+		}
+	}
+
+	newLocalRootDevice := containerGetRootDiskDevice(req.Devices)
+
+	if len(profileRootDiskDevices) > 1 && newLocalRootDevice == nil || (len(profileRootDiskDevices) > 1 && newLocalRootDevice["pool"] == "") {
+		return BadRequest(fmt.Errorf("Profiles specify contradicting storage pools the container belongs to."))
+	}
```

Comment:
Wouldn't this check be better suited for containerCreateInternal which is where we already run all the other config checks?